src/Renderer/ThreeExtended/Feature2Mesh.js

```diff
@@ -1,90 +1,338 @@
+}
+
+/*
+ * Add indices for the side faces.
+ * We loop over the contour and create a side face made of two triangles.
+ *
+ * For a contour made of (n) coordinates, there are (n*2) vertices.
+ * The (n) first vertices are on the roof, the (n) other vertices are on the floor.
+ *
+ * If index (i) is on the roof, index (i+length) is on the floor.
+ *
+ * @param {number[]} indices - Indices of vertices
+ * @param {number} length - length of the contour of the feature
+ * @param {number} offset - index of the first vertice of this feature
+ */
+function addFaces(indices, length, offset) {
+    // loop over contour length, and for each point of the contour,
+    // add indices to make two triangle, that make the side face
+    for (let i = offset; i < offset + length - 1; ++i) {
+        // first triangle indices
+        indices.push(i);
+        indices.push(i + length);
+        indices.push(i + 1);
+        // second triangle indices
+        indices.push(i + 1);
+        indices.push(i + length);
+        indices.push(i + length + 1);
+    }
+}
+
+function coordinateToPoint(coordinates, properties, options) {
+    let vertices = [];
+    let colors = [];
+    const geometry = new THREE.BufferGeometry();
+
+    /* eslint-disable guard-for-in */
+    for (const id in coordinates.featureVertices) {
+        const { contour, propertie } = extractFeature(coordinates, properties, id);
+        // get altitude from properties
+        const altitude = getAltitude(options, propertie);
+        const newVertices = coordinatesToVertice(contour, altitude);
+        vertices = vertices.concat(newVertices);
+
+        // assign color to each point
+        const colorArray = createColorArray(contour.length, getColor(options, propertie), 255);
+        colors = colors.concat(colorArray);
+    }
+
+    geometry.addAttribute('position', new THREE.BufferAttribute(new Float32Array(vertices), 3));
+    return new THREE.Points(geometry);
+}
+
+function coordinateToLine(coordinates, properties, options) {
+    let vertices = [];
+    const indices = [];
+    let colors = [];
+    const geometry = new THREE.BufferGeometry();
 
-    geometry.addAttribute('position', new THREE.BufferAttribute(vertices, 3));
+    /* eslint-disable guard-for-in */
+    for (const id in coordinates.featureVertices) {
+        const { contour, propertie } = extractFeature(coordinates, properties, id);
+        // get altitude from properties
+        const altitude = getAltitude(options, propertie);
+        const newVertices = coordinatesToVertice(contour, altitude);
+        vertices = vertices.concat(newVertices);
 
-    // build indice and instanciate mesh
-    let result;
+        // set indices
+        const line = coordinates.featureVertices[id];
+        // TODO optimize indices lines
+        // is the same array each time
+        for (let i = line.offset; i < line.offset + line.count - 1; ++i) {
+            indices.push(i);
+            indices.push(i + 1);
```

Comment:
It seems you could just do `indices.push(i)` use a `THREE.Line` (instead of `LineSegments`).
Any reason for not doing this?

Reply:
But you'd have to create 1 `Line` per feature, instead of a 1 `LineSegment`.

Reply:
> Any reason for not doing this?
It is the existing code, I din't change it, just refactor.

Reply:
You can change the way we create lines, changing LineSegment to Line if you want in another PR, but it's not the purpose of this PR. This PR is about polygon extrusion.

Reply:
Good thing that I didn't asked you to change this then :)